compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -764,40 +807,69 @@ def list_endpoints():
 
 
 @app.command("delete")
-@name_or_uuid_arg
+@uuid_arg_or_local_dir
 @click.option(
     "--force",
     default=False,
     is_flag=True,
-    help="Deletes the local endpoint even if the web service returns an error.",
+    help=(
+        "Ignores any errors encountered while attempting to delete an "
+        "endpoint locally and remotely."
+    ),
 )
 @click.option(
-    "--yes", default=False, is_flag=True, help="Do not ask for confirmation to delete."
+    "--yes",
+    default=False,
+    is_flag=True,
+    help="Do not ask for confirmation to delete endpoints.",
 )
 @common_options
 @handle_auth_errors
-def delete_endpoint(*, ep_dir: pathlib.Path, force: bool, yes: bool):
+def delete_endpoint(
+    *,
+    ep_dir: pathlib.Path | None,
+    ep_uuid: str | None = None,
+    force: bool,
+    yes: bool,
+):
     """Deletes an endpoint and its config."""
 
-    ep_conf = None
-    try:
-        ep_conf = get_config(ep_dir)
-    except Exception as e:
-        print(f"({type(e).__name__}) {e}\n")
-        if not yes:
-            yes = click.confirm(
-                f"Failed to read configuration from {ep_dir}/\n"
-                f"  Are you sure you want to delete endpoint <{ep_dir.name}>?",
-                abort=True,
-            )
+    ep_info = f"< {ep_dir.name} >" if ep_dir else ""
+    if ep_uuid:
+        ep_info += f" ( {ep_uuid} )"
 
     if not yes:
-        yes = click.confirm(
-            f"Are you sure you want to delete endpoint <{ep_dir.name}>?",
+        click.confirm(
+            f"Are you sure you want to delete endpoint {ep_info}?",
             abort=True,
         )
 
-    Endpoint.delete_endpoint(ep_dir, ep_conf, force=force)
+    ep_conf = None
+    error_msg = None
+    try:
+        if ep_dir:
+            ep_conf = get_config(ep_dir)
+            if ep_uuid is None:
+                ep_uuid = Endpoint.get_endpoint_id(ep_dir)
+
+        else:
+            error_msg = f"No local endpoint configuration available for {ep_uuid}"
+    except Exception:
+        # Endpoint directory exists locally but error loading the config
+        error_msg = f"Failed to read configuration from {ep_dir}/ "
```

Comment:
Suggested minor refactor for clarity:
```suggestion
if ep_dir:
    try:
        ep_conf = get_config(ep_dir)
        if ep_uuid is None:
            ep_uuid = Endpoint.get_endpoint_id(ep_dir)
    except Exception:
        # Endpoint directory exists locally but error loading the config
        error_msg = f"Failed to read configuration from {ep_dir}/ "
else:
    error_msg = f"No local endpoint configuration available for {ep_uuid}"
```